docs(wasm): Add details about project structure in Rust to Wasm tutorial (#27025)
mdn/content: files/en-us/webassembly/rust_to_wasm/index.md

> Let's start by creating a file named `index.html` in the root of the project, and give it the following contents:
> 
> Project Structure
> 
> ```plain
> ├── Cargo.lock
> ├── Cargo.toml
> ├── index.html
> ├── pkg
> │   ├── hello_wasm.d.ts
> │   ├── hello_wasm.js
> │   ├── hello_wasm_bg.wasm
> │   ├── hello_wasm_bg.wasm.d.ts
> │   └── package.json
> ├── src
> │   └── lib.rs
> └── target
>     ├── CACHEDIR.TAG
>     ├── release
>     └── wasm32-unknown-unknown
> ```
> 

Reviewer: ````suggestion
Let's start by creating a file named `index.html` in the root of the project, so we end up with the following project structure:

```plain
├── Cargo.lock
├── Cargo.toml
├── index.html
├── pkg
│   ├── hello_wasm.d.ts
│   ├── hello_wasm.js
│   ├── hello_wasm_bg.wasm
│   ├── hello_wasm_bg.wasm.d.ts
│   └── package.json
├── src
│   └── lib.rs
└── target
    ├── CACHEDIR.TAG
    ├── release
    └── wasm32-unknown-unknown
```

Put the following content in the `index.html` file:

````

This imports the new module from the `node_modules` folder. This isn't considered a best practice, but this is a demo, so it's OK for now. Once it's loaded, it calls the `greet` function from that module, passing `"WebAssembly"` as a string. Note how there's nothing special here, yet we're calling into Rust code. As far as the JavaScript code can tell, this is just a normal module.

> Finally, we need to modify the HTML file, which was created under the `pkg` folder; open the `index.html` file and replace the current contents with the following:

Reviewer 2: We only have one `index.html` file mentioned in the doc, so this is clear for me.
Alternatively, you could say:

```suggestion
Finally, we need to modify the HTML file located at `../pkg/index.html`.
Open the `index.html` file and replace the current contents with the following:
```

Author: Hi @bsmth ,

Thanks for the suggestion. I have tried to implement it. I would request you to co-author this PR with me. It would be an learning experience for me.

Reviewer 2: Sounds good, I'll take a look at this again shortly to see if I can follow it from the beginning from scratch to reproduce the steps.

Reviewer 2: @postmeback - I've made some changes based on trying out the steps in the tutorial, it's working as expected for me.

Author: Okay, if everything seems fine. We should merge it.

This imports the new module from the `node_modules` folder. This isn't considered a best practice, but this is a demo, so it's OK for now. Once it's loaded, it calls the `greet` function from that module, passing `"WebAssembly"` as a string. Note how there's nothing special here, yet we're calling into Rust code. As far as the JavaScript code can tell, this is just a normal module.

Finally, we need to modify the HTML file, which was created under the `pkg` folder.
Our project should look like this:

> ```plain

Reviewer: ````suggestion
```
````
